packages/medusa/src/services/cart.ts

```diff
@@ -1223,8 +1276,39 @@ class CartService extends TransactionBaseService {
           isDefined(data.sales_channel_id) &&
           data.sales_channel_id != cart.sales_channel_id
         ) {
+          const salesChannel = await this.getValidatedSalesChannel(
+            data.sales_channel_id
+          )
+
           await this.onSalesChannelChange(cart, data.sales_channel_id)
-          cart.sales_channel_id = data.sales_channel_id
+
+          /**
+           * TODO: remove this once update cart workflow is build
+           * since this will be handled in a handler by the workflow
+           */
+          if (this.featureFlagRouter_.isFeatureEnabled(MedusaV2Flag.key)) {
+            if (cart.sales_channel_id) {
+              await this.remoteLink_.dismiss({
+                cartService: {
+                  cart_id: cart.id,
+                },
+                salesChannelService: {
+                  sales_channel_id: cart.sales_channel_id,
+                },
+              })
+            }
+
+            await this.remoteLink_.create({
+              cartService: {
+                cart_id: cart.id,
+              },
+              salesChannelService: {
+                sales_channel_id: salesChannel.id,
+              },
+            })
```

Comment:
question: would it work concurrently?

Reply:
there is a unique constraint on the `cart_id` column